code/game/machinery/pipe/construction.dm

```diff
@@ -167,23 +167,29 @@ Buildable meters
 			to_chat(user, span_warning("Something is hogging the tile!"))
 			return TRUE
 
-		if(pipe_count == 1 && istype(machine, /obj/machinery/atmospherics/pipe/smart) && ispath(pipe_type, /obj/machinery/atmospherics/pipe/smart) && lowertext(machine.pipe_color) != lowertext(pipe_color) && machine.connection_num < 3)
-			var/direction = machine.dir
-			if((direction & EAST|WEST || direction & SOUTH|NORTH) && !ISDIAGONALDIR(direction))
-				pipe_type = /obj/machinery/atmospherics/pipe/bridge_pipe
-				if(EWCOMPONENT(direction))
-					dir = NORTH
-				if(NSCOMPONENT(direction))
-					dir = EAST
-				continue
-
-		if(flags & PIPING_BRIDGE && !(machine.pipe_flags & PIPING_BRIDGE) && check_ninety_degree_dir(machine)) //continue if we are placing a bridge pipe over a normal pipe only (prevent duplicates)
+		// skip checks if we don't overlap layers, either by being on the same layer or by something being on all layers
+		if((machine.piping_layer != piping_layer) && !((machine.pipe_flags | flags) & PIPING_ALL_LAYER))
 			continue
 
-		if((machine.piping_layer != piping_layer) && !((machine.pipe_flags | flags) & PIPING_ALL_LAYER)) //don't continue if either pipe goes across all layers
+		// If we're a smart pipe, and we're looking to share space with a potentially perpendicular kind of pipe, see if we can smartly go over it as a bridge
+		if(pipe_count == 1 && machine.connection_num < 3 && ispath(pipe_type, /obj/machinery/atmospherics/pipe/smart))
+			// If we're looking to go over another smart pipe, it must be of a different color.
+			if (!istype(machine, /obj/machinery/atmospherics/pipe/smart) || lowertext(machine.pipe_color) != lowertext(pipe_color))
+				var/other_direction = machine.dir
+				if(((other_direction & (EAST|WEST)) || (other_direction & (SOUTH|NORTH))) && !ISDIAGONALDIR(other_direction))
+					pipe_type = /obj/machinery/atmospherics/pipe/bridge_pipe
```

Comment:
Since you can now set the direction of the pipes I think you could ditch this behaviour but uh test it before doing it

Reply:
Tho this may not work with mapped pipes, at least because my pr to set the dir of mapped pipes didn't pass

Reply:
Line 60 in the first commit ( https://github.com/tgstation/tgstation/pull/60981/commits/631cafe48e44dce15c82dd81326714504c203bd8#diff-8aade545476ce46e983e1a3fa0bfd6e4d2cfcee4490b8652ed644ee5044b41feR60 ) should fix dir problems in a similar way to #60804 but more concisely. I've tested and confirmed that this works with mapped pipes now

Reply:
I tried to stick to fairly conservative fixes for this PR, since the current situation is a pain in the ass.

I'll look into it anyway, though. If this PR is still open once I'm confident that behavior removal doesn't break anything (which might take a while there are a shitload of edge cases dear god) I'll amend it with some more cleanup. Otherwise, I'll follow up in another PR.

Reply:
Hmm, by being able to set the direction of the pipes now, did you mean that bridges could be replaced by a "regular" pipe going straight across? I think you might be able to smartly build a corner when there's only an opposing corner on the same layer, in a similar fashion. I'll see how that feels and will probably keep that for another PR though

Reply:
IIRC you can in theory have the pipes cross each other if their direction is different, like you can have a + cross with one pipe N|S and the other E|W, you can also have two corners on the same layer on the same tile too.
You could try to remove the auto bridge pipe thing, but you'll have to test it a bit

Reply:
Your command is my wish